core/src/test/java/com/graphhopper/routing/util/WheelchairFlagEncoderTest.java

```diff
@@ -548,4 +546,44 @@ public void testSmoothness() {
         way.setTag("sidewalk:both:smoothness", "horrible");
         assertTrue(wheelchairEncoder.getAccess(way).canSkip());
     }
+
+    private Graph initExampleGraph() {
+        GraphHopperStorage gs = new GraphHopperStorage(new RAMDirectory(), encodingManager, true, new GraphExtension.NoOpExtension()).create(1000);
+        NodeAccess na = gs.getNodeAccess();
+        // incline of 5% over all
+        na.setNode(0, 51.1, 12.001, 50);
+        na.setNode(1, 51.1, 12.002, 55);
+        EdgeIteratorState edge0 = gs.edge(0, 1).setWayGeometry(Helper.createPointList3D(51.1, 12.0011, 49, 51.1, 12.0015, 55));
+        edge0.setDistance(100);
+        GHUtility.setProperties(edge0, wheelchairEncoder, 5, 5);
+
+        // incline of 10% over all
+        na.setNode(2, 51.2, 12.101, 50);
+        na.setNode(3, 51.2, 12.102, 60);
+        EdgeIteratorState edge1 = gs.edge(2, 3).setWayGeometry(Helper.createPointList3D(51.2, 12.1011, 49, 51.2, 12.1015, 55));
+        edge1.setDistance(100);
+        GHUtility.setProperties(edge1, wheelchairEncoder, 5, 5);
+        return gs;
+    }
+
+    @Test
+    public void testApplyWayTags() {
+        Graph graph = initExampleGraph();
+
+        EdgeIteratorState edge0 = GHUtility.getEdge(graph, 0, 1);
+        ReaderWay way0 = new ReaderWay(1);
+        wheelchairEncoder.applyWayTags(way0, edge0);
+
+        IntsRef flags0 = edge0.getFlags();
+        assertEquals(2, wheelchairEncoder.getSpeed(false, flags0), 1e-1);
```

Comment:
Instead of the encoder use the DecimalEncodedValue:

```
edge0.get(encoder.getAverageSpeedEnc())
edge0.getReverse(encoder.getAverageSpeedEnc())
```